src/main/webapp/app/modules/canvas/visualization/util/visualmetadata-wrapper.ts

```diff
@@ -0,0 +1,294 @@
+};
+
+const isDimension = item => {
+  return item.feature && item.feature != null && item.feature.featureType === 'DIMENSION';
+};
+
+const isMeasure = item => {
+  return item.feature && item.feature != null && item.feature.featureType === 'MEASURE';
+};
+
+const getDimension = item => {
+  return item.sort(function (a, b) {
+    return a.fieldType.order - b.fieldType.order;
+  });
+};
+
+const constructMeasureField = fieldMeasure => {
+  const agg = getProperty(fieldMeasure.properties, 'Aggregation type', null);
+  if (agg !== null) {
+    return {
+      aggregation: agg !== 'NONE' ? agg : null,
+      name: fieldMeasure.feature.definition,
+      alias: fieldMeasure.feature.name,
+    };
+  } else {
+    return {
+      name: fieldMeasure.feature.definition,
+    };
+  }
+};
+
+const constructHavingField = fieldMeasure => {
+  const agg = getProperty(fieldMeasure.properties, 'Aggregation type', null);
+  if (agg !== null && agg !== 'NONE') {
+    return { name: fieldMeasure.feature.definition, aggregation: agg };
+  }
+  return null;
+};
+
+const getQueryParametersWithFields = (fields, filters, conditionExpression) => {
+  var query = {
+    fields,
+  };
+
+  // if (conditionExpression && conditionExpression.conditionExpression) {
+  //   query.conditionExpressions = [conditionExpression];
+  // }
+
+  return query;
+};
+
+const getQueryParameters = (visual, filters, conditionExpression, offset) => {
+  const fields = visual.fields;
+  const dimensions = fields.filter(isDimension);
+
+  const measures = fields.filter(isMeasure);
+
+  const dimensionFields = dimensions.map(function (item) {
+    const result = constructDimensionField(item, filters);
+    item.feature.selectedName = result.name;
+    return result;
+  });
+
+  const measureFields = measures.map(function (item) {
+    return constructMeasureField(item);
+  });
+
+  var query;
+
+  query = getQueryParametersWithFields(dimensionFields.concat(measureFields), filters, conditionExpression);
+
+  if (visual.metadataVisual.name === 'Table' || visual.metadataVisual.name === 'Pivot Table') {
+    query.offset = getChartPropertyValue('Limit', 20) * offset;
+  }
+
+  const aggExists = !!measureFields.filter(function (item) {
+    return item.aggregation;
+  })[0];
+
+  if (aggExists) {
+    query.groupBy = dimensionFields;
+  }
+  query.limit = getChartPropertyValue(visual.properties, 'Limit', 20);
+  const ordersListSortMeasures = measures
+    .filter(function (item) {
+      const property = getProperty(item.properties, 'Sort', null);
+      return property !== null && property !== 'None';
+    })
+    .map(function (item) {
+      const property = getProperty(item.properties, 'Sort', null);
+      if (property === 'Ascending') {
+        return {
+          direction: 'ASC',
+          feature: { name: item.feature.name },
+        };
+      } else {
+        return {
+          direction: 'DESC',
+          feature: { name: item.feature.name },
+        };
+      }
+    });
```

Comment:
put all const in separate file..this is a big function...create small-2 function

Reply:
done